app/controllers/user_controller.php

```diff
@@ -73,4 +84,98 @@ public function logout()
         session_destroy();
         redirect('user/login');
     }
+
+    public function profile()
+    {
+        $user = new User();
+        $user->id = Param::get('user_id');
+        $user->getProfile();
+
+        if(!isset($user->username)) {
+            redirect('notfound/pagenotfound');
+        }
+
+        $threads_followed = array();
+        $thread_followed_id = Follow::getFollowedThreadIds($user->id);
+
+        foreach($thread_followed_id as $thread) {
+            $threads_followed[] = Thread::getById($thread['thread_id']);
+        }
+
+        foreach ($threads_followed as $thread) {
+            $thread->username = User::getUsernameById($thread->user_id);
+        }
+
+        $threads_created = Thread::getByUserId($user->id);
+
+        foreach ($threads_created as $thread) {
+            $thread->username = User::getUsernameById($thread->user_id);
+        }
+
+        $this->set(get_defined_vars());
+    }
+
+    public function edit()
+    {
+        $process = Param::get('process', 'edit');
+        $user = new User();
+
+        switch($process) {
+            case self::EDIT_ACCOUNT:
+                $user->id = $_SESSION['userid'];
```

Comment:
Refactored my code po. I wrapped the  process po of authenticating the user id in my helper function so I will not have to repetitively write try and catch po. https://github.com/renzosunico/MyClassroom/commit/b38376515c72c296cf46c444804d9cd9c0324ba9